src/main/scala/com/github.sstone/amqp/ChannelOwner.scala

```diff
@@ -51,10 +60,9 @@ object ChannelOwner {
           def handleFlow(active: Boolean): Unit = listener ! HandleFlow(active)
         }))
       }
-      case request@Publish(exchange, routingKey, body, properties, mandatory, immediate) => {
+      case request@Publish(_, _, _, _, _, _) => {
```

Comment:
yeah, unfortunately it compiles to the same bytecode, which is a huge downer. here's an article on this that Chris pointed me to: http://typelevel.org/blog/2014/11/10/why_is_adt_pattern_matching_allowed.html

if you guys feel like the code becomes significantly less readable by the constructor pattern (and the @ binding), then let's use the typed pattern